kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -440,7 +440,13 @@ impl<'tcx> GotocCtx<'tcx> {
                     | InstanceDef::ReifyShim(..)
                     | InstanceDef::ClosureOnceShim { .. }
                     | InstanceDef::CloneShim(..) => {
-                        let func_exp = self.codegen_operand(func);
+                        // We need to codegen the operand normally even though we discard the result
+                        // because we need to ensure the symbol table is updated correctly.
+                        // (When deleted, the function `lang_start_internal` is not found anymore when running `cargo kani`.)
+                        let _ = self.codegen_operand(func);
+                        // We need to handle FnDef items specially because we compile them to dummy structs.
```

Comment:
Where should I put all these explanations? It would be good if we had one place where we describe how we compile certain language constructs instead of scattered comments.

Reply:
I've now added a link to this PR. I think it gives the most context for the changes.